posts/2024-02-12-Java21-Images.adoc

```diff
@@ -0,0 +1,40 @@
+---
+layout: post
+title: "TITLE"
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/tjwatson
+author_github: https://github.com/tjwatson
+seo-title: Liberty container images with Semeru Runtime Java 21 support
+seo-description: Liberty container images now offer Java 21 support using the Semeru Runtime JRE. The Java 21 images use UBI 9 minimal base images and the JRE for the production of smaller application images. The Java 21 UBI 9 images now offer InstantOn support for the Power and Z (s390x) processor architectures along with the x86-64/AMD64 process architectures. 
+blog_description: "DESCRIPTION"
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+open-graph-image-alt: Open Liberty Logo
+---
+= Liberty container images with Semeru Runtime Java 21 support
+Thomas Watson <https://github.com/tjwatson>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+
+The high performance link:https://developer.ibm.com/languages/java/semeru-runtimes[IBM Semeru Runtime] has released support for link:https://community.ibm.com/community/user/wasdevops/blogs/tony-renaud/2024/01/11/ibm-semeru-runtime-open-edition-2101-is-out-today[Java 21]. The various Java 21 language features are now supported such as Virtual threads, scoped values, record patterns, the latest foreign functions and memory API and more. For Java 21, the Open Liberty project has moved to using UBI 9 minimal images with the Semeru Runtime Java 21 JRE link:https://github.com/ibmruntimes/semeru-containers[container] images.
+
+== Universal Base Image Minimal Version 9
+The link:https://catalog.redhat.com/software/containers/ubi9/ubi-minimal/615bd9b4075b022acc111bf5[Universal Base Image 9 Minimal] is a stripped down image that uses `microdnf` as a package manager which allows for the production of smaller application images. Open Liberty Java 21 UBI 9 minimal link:https://github.com/OpenLiberty/ci.docker/blob/main/docs/icr-images.md[images] are available starting with the 24.0.0.1 release of Open Liberty. The UBI 9 minimal content may impact some application image builds because of missing functionality. Application image builds can use `microdnf` to install additional packages as required. At the time of this publication UBI 9 is not FIPS compliant. Other Open Liberty Java version images will continue to use the UBI 8 standard images (Java 8, Java 11, Java 17).
+
+== Semeru Runtime Java 21 JRE
+The Open Liberty Java 21 images use the Semeru Runtime JRE image as the Java 21 runtime. The Semeru Runtime JRE images are considerably smaller than the JDK images. The Liberty Java 21 images provide a very nice reduction in the size of application images due to the use of the UBI 9 minimal image and the Semeru Runtime JRE image. The reduced content of the JRE image may impact some application images because of its reduced functionality.
```

Comment:
```suggestion
The high-performance link:https://developer.ibm.com/languages/java/semeru-runtimes[IBM Semeru Runtime] now supports link:https://community.ibm.com/community/user/wasdevops/blogs/tony-renaud/2024/01/11/ibm-semeru-runtime-open-edition-2101-is-out-today[Java 21], including the various Java 21 language features, such as Virtual threads, scoped values, record patterns, the latest foreign functions and memory API, and more. 

For Java 21, the Open Liberty project now uses  UBI 9 minimal images with the Semeru Runtime Java 21 JRE link:https://github.com/ibmruntimes/semeru-containers[container] images. The Semeru Runtime JRE images are considerably smaller than the JDK images. The Liberty Java 21 images provide a very nice reduction in the size of application images due to the use of the UBI 9 minimal image and the Semeru Runtime JRE image. The reduced content of the JRE image might impact some application images because of its reduced functionality.
```